packages/wallet-core/src/bignumber.ts

```diff
@@ -0,0 +1,52 @@
+import {BigNumber as EthersBigNumber, BigNumberish} from 'ethers';
+import {Uint256} from './types';
+
+type T = BigNumberish;
+
+const binaryOperator = (name: keyof EthersBigNumber) => (a: T, b: T): Uint256 => {
+  if (typeof EthersBigNumber.from(a)[name] !== 'function') throw Error(`Invalid method ${name}`);
+
+  const result = (EthersBigNumber.from(a)[name] as any)(b);
+
+  return EthersBigNumber.isBigNumber(result) ? result.toHexString() : result;
+};
+
+const unaryOperator = <S extends any = Uint256>(name: keyof EthersBigNumber) => (a: T): S => {
+  if (typeof EthersBigNumber.from(a)[name] !== 'function') throw Error(`Invalid method ${name}`);
+
+  const result = (EthersBigNumber.from(a)[name] as any)();
+  return EthersBigNumber.isBigNumber(result) ? result.toHexString() : result;
+};
+
+/**
+ * This is a convenience class that looks like instances of ether's BigNumber class.
+ * This way, people can call
+ * ```
+ * BN.eq(2, '3') // false
+ * BN.add('0x123', 4) // '0x661'
+ * BN.add('0xabc', 10) // '0xac6'
+ */
+
+export class BigNumber {
```

Comment:
Naming this `BigNumber` may confuse it with ether's `BigNumber` (especially when vscode adds the import statement). Maybe we could go with something slightly different so we don't run into that?

Reply:
What about `BN`?

Reply:
There is a `BN.js` [library](https://github.com/indutny/bn.js/) but that's probably fine since we don't use it, so I'd be good with `BN`